src/wp-includes/option.php

```diff
@@ -1508,6 +956,24 @@ function get_transient( $transient ) {
  */
 function set_transient( $transient, $value, $expiration = 0 ) {
 
+	// Transient name should not be empty string, 0, or false.
+	if ( empty( $transient ) ) {
```

Comment:
If you do something like this, I think you can remove the `true` and `is_scalar` checks below. Transient name should be a string, as documented, and as noted by @hellofromtonya. 

You could also remove the cast to string on line 970, since it would have already been confirmed as a string.

```suggestion
	if ( ! is_string( $transient ) || empty( $transient ) ) {
```